actions/command.js

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Client4} from 'mattermost-redux/client';
+import {unfavoriteChannel} from 'mattermost-redux/actions/channels';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentRelativeTeamUrl} from 'mattermost-redux/selectors/entities/teams';
+
+import {openModal} from 'actions/views/modals';
+import * as GlobalActions from 'actions/global_actions.jsx';
+import * as PostActions from 'actions/post_actions.jsx';
+
+import {isUrlSafe, getSiteURL} from 'utils/url.jsx';
+import {localizeMessage, getUserIdFromChannelName} from 'utils/utils.jsx';
+import * as UserAgent from 'utils/user_agent.jsx';
+import {Constants, ModalIdentifiers} from 'utils/constants.jsx';
+import {isFavoriteChannel} from 'utils/channel_utils.jsx';
+import {browserHistory} from 'utils/browser_history';
+
+import UserSettingsModal from 'components/user_settings/modal';
+
+export function executeCommand(message, args) {
+    return async (dispatch, getState) => {
+        const state = getState();
+
+        let msg = message;
+
+        let cmdLength = msg.indexOf(' ');
+        if (cmdLength < 0) {
+            cmdLength = msg.length;
+        }
+        const cmd = msg.substring(0, cmdLength).toLowerCase();
+        msg = cmd + ' ' + msg.substring(cmdLength, msg.length).trim();
+
+        switch (cmd) {
+        case '/search':
+            PostActions.searchForTerm(msg.substring(cmdLength + 1, msg.length));
```

Comment:
I'm assuming this one is not yet a redux action, right?

Reply:
Correct